lib/experimental/modules/class-gutenberg-modules.php

```diff
@@ -123,15 +123,21 @@ public static function print_module_preloads() {
 	 * import maps (https://github.com/guybedford/es-module-shims/issues/371).
 	 */
 	public static function print_import_map_polyfill() {
-		$import_map = self::get_import_map();
-		if ( ! empty( $import_map['imports'] ) ) {
-			wp_print_script_tag(
-				array(
-					'src'   => gutenberg_url( '/build/modules/importmap-polyfill.min.js' ),
-					'defer' => true,
-				)
-			);
-		}
+		$test = 'HTMLScriptElement.supports?.("importmap")';
```

Comment:
`?.` isn't well supported by older browsers as far as I know, so probably safer to avoid it here, given this is for a polyfill check. Maybe better like this:
```suggestion
		$test = 'HTMLScriptElement.supports && HTMLScriptElement.supports("importmap")';
```

Reply:
It seems to work as expected even with `?.` in Firefox 107 that doesn't support import maps:

https://caniuse.com/import-maps

The polyfill loads as expected. I see some entries on the console printer, so I need to confirm whether it actually works.

Reply:
Firefox 107 is still not that old, relatively speaking. My point is `?.` isn't always supported, so since this is about a polyfill, my take is that we should make sure the code to check is _very_ widely compatible. It's a simple change that enhances compatibility.

Reply:
I planned to refactor it as you suggested. I simply didn't explain myself clearly enough. I was focused on testing last evening 😄

Reply:
Oh ok, thanks for clarifying!

Reply:
@gziolo What about this change? Looking at https://caniuse.com/mdn-javascript_operators_optional_chaining, the operator is certainly well supported, but still not everywhere, including Internet Explorer. I don't see a huge value in using it here, primarily because it is used for a polyfill check which has the sole purpose of cross-browser support, and it only reduces the code by a few characters anyway.

Reply:
On my todo list. I didn’t find time to revisit the implementation but I fully agree with your rationale 👍👍👍

Reply:
I committed your comment. I hope it’s enough and I should really do it earlier 🙈

Reply:
@gziolo Sorry I didn't realize you simply hadn't come back to this yet - didn't mean to rush 🙃 

Thank you, that looks good to me!